pkg/lib/configreader/interface_map.go

```diff
@@ -139,6 +139,8 @@ func validateInterfaceMap(val map[string]interface{}, v *InterfaceMapValidation)
 			if err != nil {
 				return nil, errors.Wrap(err, key)
 			}
+
+			val[key] = stringToIntMap
```

Comment:
Nice! Maybe add a comment on the struct definition that this flag is recursive (even though I presume this was already the case before your fix)